wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/DefaultWorkspaceValidator.java

```diff
@@ -126,13 +128,24 @@ private void validateMachine(MachineConfig machineCfg, String envName) throws Ba
         checkArgument(!(machineCfg.getSource().getContent() == null && machineCfg.getSource().getLocation() == null),
                       "Environment " + envName + " contains machine with source but this source doesn't define a location or content");
 
+
         checkArgument(machineInstanceProviders.hasProvider(machineCfg.getType()),
                       "Type %s of machine %s in environment %s is not supported. Supported values: %s.",
                       machineCfg.getType(),
                       machineCfg.getName(),
                       envName,
                       Joiner.on(", ").join(machineInstanceProviders.getProviderTypes()));
 
+        if (machineCfg.getSource().getType().equals("dockerfile") && machineCfg.getSource().getLocation() != null) {
+            try {
+                final String protocol  = new URL(machineCfg.getSource().getLocation()).getProtocol();
+                checkArgument(protocol.equals("http") || protocol.equals("https"),
+                              "Environment " + envName + " contains machine with invalid source location");
```

Comment:
I think this message is not good enough. For instance you can add more certainty to the message by saying what exactly is wrong, i mean if we allow only http/https protocols for recipe locations then it is obviously good to add it to the exception, moreover if environment provides many machine configs then it will be also useful to add recipe location to the exception message as well.